sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -122,10 +126,20 @@ class HadoopTableReader(
     val attrsWithIndex = attributes.zipWithIndex
     val mutableRow = new SpecificInternalRow(attributes.map(_.dataType))
 
-    val deserializedHadoopRDD = hadoopRDD.mapPartitions { iter =>
+    val deserializedHadoopRDD = hadoopRDD.mapPartitionsWithIndex { (index, iter) =>
       val hconf = broadcastedHadoopConf.value.value
       val deserializer = deserializerClass.newInstance()
       deserializer.initialize(hconf, tableDesc.getProperties)
+      if (skipHeaderLineCount > 0 && isTextInputFormatTable) {
+        val partition = hadoopRDD.partitions(index).asInstanceOf[HadoopPartition]
```

Comment:
I wonder if it's best to be defensive and test whether these things are actually instances of the classes they're expected to be? or is that too paranoid?